src/db.rs

```diff
@@ -187,6 +192,28 @@ impl DbInner {
         }
     }
 
+    async fn maybe_apply_backpressure(&self) {
+        loop {
+            let table = {
+                let guard = self.state.read();
+                let state = guard.state();
+                if state.imm_memtable.len() <= self.options.max_unflushed_memtable {
+                    return;
+                }
+                let Some(table) = state.imm_memtable.back() else {
+                    return;
+                };
+                warn!(
+                    "applying backpressure to write by waiting for imm table flush. imm tables({}), max({})",
```

Comment:
Still reviewing, not done with all the files.
Is the part where back pressure must be applied covered in any of the tests?

Reply:
For such tests, having an explicit `compact()` function is easier to control than the timer based triggers.

Reply:
That's fair - I missed adding test coverage for this. I've just tested it manually by verifying that backpressure is applied when I load the database. I like your suggested approach - I'll give that a shot.

Reply:
@vigneshc I added a test case. LMK if it makes sense.